File: codeforces/25.george_and_accommodation/gaa.py
# Author: José Rodolfo (jric2002)
n = int(input())
available_rooms = 0
while (n):
  p, q = list(map(int, input().split(" ")))
  if ((q - p) >= 2):
    available_rooms += 1
  n -= 1
print(available_rooms)

File: codeforces/low_difficulty/800_points/25.george_and_accommodation_[467A]/gaa.cpp
#include <iostream>
/* Author: José Rodolfo (jric2002) */
using namespace std;
int main() {
  unsigned short int n, p, q, available_rooms = 0;
  cin >> n;
  while (n--) {
    cin >> p >> q;
    if ((q - p) >= 2) {
      available_rooms++;
    }
  }
  cout << available_rooms << endl;
  return 0;
}